hack/tools/golangci-lint/go.work

```diff
@@ -1,7 +1,9 @@
 // This is a hack, but it prevents go from climbing further and trying to
 // reconcile the various deps across the "real" modules and this one.
 
-go 1.24.0
+go 1.24.2
+
+toolchain go1.24.3
```

Comment:
we should probably manually delete this for now, so it continues using our desired version

auto adding toolchain when none was previously specified, instead of only updating it, is a bug that should be fixed in go 1.25 https://github.com/golang/go/issues/65847